### ISSUE TITLE
Content Collection documentation needs to updated.

### ISSUE THREAD

**Issue by brezzy1337** [withastro/docs#10569]
### 📚 Subject area/topic

Content Collection

### 📋 Page(s) affected (or suggested, for new content)

https://docs.astro.build/en/reference/modules/astro-content/
https://docs.astro.build/en/guides/content-collections/

### 📋 Description of content that is out-of-date or incorrect

The code that worked for me for content collection is the following below. I am using astrojs/vercel which could be causing something to be off from the [documentation.]

```import { defineCollection, z } from 'astro:content';

const caseStudy = defineCollection({
    type: 'content',
    schema: z.object({
        title: z.string(),
        url: z.string(),
        pubDate: z.coerce.date(),
        updatedDate: z.coerce.date(),
        heroImage: z.string(),
        description: z.string(),
        features: z.array(z.object({
            description: z.string(),
            image: z.string()
        }))
    })
});

export const collections = {
    'case-studies': caseStudy,
};
```

For some reason when I try to use the loader and glob I get a error on my defineCollection import saying it 

```
Cannot find module 'astro:content'
```

I'm not sure if this is a documentation issue or error on my end. It would be nice to address if @astrojs/vercel does have any content collection differences in configuration, etc.

Thank you for your help! Your project is super cool :)

### 🖥️ Reproduction in StackBlitz (if reporting incorrect content or code samples)

https://github.com/brezzy1337/www-digital-space-development-.com

**Comment by ArmandPhilippot** [comment]
Hi, I could be wrong but I think the issue comes from your project, not the docs:

1. You're using Astro v2 (`"astro": "^2.10.15"`) while the current documentation is for Astro **v5**. The Content Layer API was released in v5, so I think it is expected that `loader` and `glob` does not work with v2.
2. I'm not sure `content.config.ts` can work outside of `src` (see [collection config file](https://docs.astro.build/en/guides/content-collections/#the-collection-config-file) in the documentation)

If you still have the same issue after [upgrading to Astro v5](https://docs.astro.build/en/upgrade-astro/) and moving your content configuration file, you can try `pnpm astro sync`. You might need to delete the `.astro` directory before.
Depending on the progress of your project, it could be easier to just [recreate a new project](https://docs.astro.build/en/install-and-setup/).

Out of curiosity, how did you install Astro? It seems weird to see a new project that uses Astro v2.

**Comment by brezzy1337** [comment]
Hey! Yeah I used a Vercel Astro template! It would make sense that the Vercel template would be the one out of date. 

https://vercel.com/new/brezzy1337s-projects/templates/astro/astro-edge-functions

Thanks for your insight I'll reach out the the author of that template to get it updated.

**Comment by ArmandPhilippot** [comment]
Oh I understand better now! Yeah, this template uses outdated packages... if you are more comfortable with a template, the one maintained by Vercel might be a better start!

Otherwise, the first template doesn't seem to do much. You could run this two commands to init a new project [with Vercel](https://docs.astro.build/en/guides/integrations-guide/vercel/) and get to the same state I think (if you put the demo page aside):
1. `pnpm create astro@latest`
2. `pnpm astro add vercel`

Happy New Year's Eve to you too, and good luck with your project!

**Comment by brezzy1337** [comment]
Yeah I switched over to the template ran by Vercel but still having the same issue with the 'astro:content' module throwing this error:  
```
Cannot find module 'astro:content' or its corresponding type declarations.ts(2307)
```

I know that the typechecking is really strict with astro so my Collection could be defined improperly.

```
import { defineCollection, z } from 'astro:content';
import { glob } from 'astro/loaders';

const caseStudy = defineCollection({
    loader: glob({ pattern: '*.md', base: 'src/content/case-studies' }),
    schema: z.object({
        title: z.string(),
        url: z.string(),
        desciption: z.string(),
        datePosted: z.coerce.date(),
        dateUpdated: z.coerce.date(),
        features: z.array({
            feature: z.object({
                title: z.string(),
                desciption: z.string(),
            }),
        }),
    }),
});

export const collections = {
    'case-studies': caseStudy,
};
```

Here's my new repo on git: https://github.com/brezzy1337/digital-space-development

Hopefully I can get to the bottom of this pretty quickly, and come back with a solution :)

**Comment by brezzy1337** [comment]
Okay got it to work here where my steps:

1. Made sure I had a .md file filled out

2. Added typescript config, which should be inherent with the template, same with tailwinds. 

3. Also installing the astro vs code extension is needed which should be on the templates documentation.
